src/configuration/configuration.ts

```diff
@@ -472,6 +470,16 @@ class Configuration implements IConfiguration {
   visualModeKeyBindingsMap: Map<string, IKeyRemapping> = new Map();
   commandLineModeKeyBindingsMap: Map<string, IKeyRemapping> = new Map();
 
+  get textwidth(): number {
```

Comment:
Is the getter necessary? These properties *should* get updated when the setting changes

Reply:
I just tested this out locally, it looks like that is true but the property is cached based on the most recently updated value, at least based on my limited testing.

e.g. If you update Ruby to have a textwidth of 20 but the global value (default or configured value) is 80, non-Ruby files will start wrapping at 20 since the value is cached and not dynamically fetched. By dynamically fetching the value the VSCode API seems to be returning the correct settings for the language based on the active file. 

These API's are all new to me so I'm not 100% certain if this is the best way to accomplish this but it seems like it's working as intended.

Reply:
Ah, got it, thanks. The whole config system probably needs a deeper refactor to allow easily changing `scope` for other settings, but this looks like a step in the right direction.